warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctions.java

```diff
@@ -145,6 +146,29 @@ public static FunctionalSet<ValueTuple> isNotNull(Object fieldValue) {
                 return FunctionalSet.singleton(getHitTerm(fieldValue));
             }
         }
+        
+        return FunctionalSet.emptySet();
+    }
+    
+    /**
+     * Returns a {@link FunctionalSet} of hit terms found for {@code fieldValue1}. If {@code fieldValue1} is a singular value tuple, a singleton
+     * {@link FunctionalSet} with the hit term from it will be returned. If {@code fieldValue1} and {@code fieldValue2} are non-empty collections of value
+     * tuples, a {@link FunctionalSet} containing the hit terms from each value in the {@code fieldValue1} collection will be returned. Otherwise, an empty
+     * {@link FunctionalSet} will be returned.
+     *
+     * @param fieldValue1
+     *            1st field value to evaluate.
+     * @param fieldValue2
+     *            2nd field value to evaluate.
+     * @return the {@link FunctionalSet} of hit terms found
+     */
+    public static FunctionalSet<ValueTuple> isNotNull(Object fieldValue1, Object fieldValue2) {
+        FunctionalSet<ValueTuple> matches = isNotNull(fieldValue1);
+        
+        if (matches != FunctionalSet.EMPTY_SET && !((FunctionalSet) isNotNull(fieldValue2)).isEmpty()) {
```

Comment:
It might be prudent to use ! FunctionalSet.EMPTY_SET.equals(matches) instead of !=.

Reply:
There have been significant changes from the initial MR, including the unit tests. Can you please re-review when you have a chance. Thanks!